src/main/scala/io/github/kiemlicz/shelm/ChartMuseumClient.scala

```diff
@@ -0,0 +1,139 @@
+package io.github.kiemlicz.shelm
+
+import io.github.kiemlicz.shelm.ChartRepositoryAuth.{Bearer, Cert, NoAuth, UserPassword}
+import io.github.kiemlicz.shelm.exception.HelmPublishException
+import sbt.util.Logger
+import sbt.{Artifact, PublishConfiguration}
+
+import java.io.{File, FileInputStream}
+import java.net.http.HttpResponse.BodyHandlers
+import java.net.http.{HttpClient, HttpRequest, HttpResponse}
+import java.time.Duration
+import java.util.Base64
+import java.util.concurrent.CompletableFuture
+
+class ChartMuseumClient(httpClient: HttpClient, requestTimeout: Duration, pushRetries: Int = 1) {
```

Comment:
I'm confused by this approach: throughout the source code, at some places we just outsource the work to the external `helm` process, and at other places we reimplement the "helm" client program in Scala.

Is it the long-term goal to stop relying on the extenal `helm` process?
If not, then why are we rewriting the `helm publish` in Scala?

Reply:
Please see the last comment